src/main/scala/introprog/IO.scala

```diff
@@ -47,7 +51,34 @@ object IO:
     * @param enc the encoding of the file.
     * */
   def saveLines(lines: Seq[String], fileName: String, enc: String = "UTF-8"): Unit =
-    saveString(lines.mkString("\n"), fileName, enc)
+    saveString(lines.mkString("\n") + "\n", fileName, enc)
+
+  /**
+    * Appends `string` to the text file `fileName` using encoding `enc`.
+    *
+    * @param text the text to be appended to the file.
+    * @param fileName the path of the file.
+    * @param enc the encoding of the file.
+    * */
+  def appendString(text: String, fileName: String, enc: String = "UTF-8"): Unit =
+    val f = new java.io.File(fileName);
+    require(!f.isDirectory(), "The file you're trying to write to can't be a directory.")
+    val w =
+      if f.exists() then
+        new BufferedWriter(new FileWriter(fileName, Charset.forName(enc), true))
+      else
+        new java.io.PrintWriter(f, enc)
+    try w.write(text) finally w.close()
+
+  /**
+    * Appends `lines` to the text file `fileName` using encoding `enc`.
+    *
+    * @param lines the lines to append to the file.
+    * @param fileName the path of the file.
+    * @param enc the encoding of the file.
+    * */
+  def appendLines(lines: Seq[String], fileName: String, enc: String = "UTF-8"): Unit =
+    appendString(lines.mkString("\n"), fileName, enc)
```

Comment:
also here it should be `mkString("","\n","\n")` I think, but check that it works as expected.

Reply:
I'll merge this and then fix the correct behavior wen empty in a later commit.